sdk/simulator-service/src/implementation/simulator-engine/simulator.ts

```diff
@@ -23,23 +25,49 @@ export class Simulator<
   }
 
   static create<S extends SimulationStrategy>(schema: S) {
-    return new Simulator(schema)
+    return new Simulator(schema, schema)
   }
 
   public async run(): Promise<SimulationState> {
-    const getReference = (path: [string, string]) => {
-      const [step, name] = path
-      return {
+    for (let i = 0; i < this.nextArray.length; i++) {
+      const proccesesedStepSchema = this.originalSchema[i]
```

Comment:
Should we check that `nextArray` and `originalSchema` have the same number of elements?

Reply:
no as nextArray gets first element removed with next step added.

Reply:
so original schema will always be at least the lenght of next Array.